tests/unittests/test_call_endpoint.py

```diff
@@ -0,0 +1,224 @@
+                        "source_info": {
+                            "ip": "10.1.39.32"
+                        },
+                        "user_match_id": "5ca22ac8e5a7400deb1ab0bd"
+                    },
+                    "type": "Event::Endpoint::NotProtected",
+                    "location": "Lightning-oxidtdinku",
+                    "description": "Failed to protect computer: Lightning-oxidtdinku",
+                    "source": "Lightning-g7n7wdv611\\Lightning"
+                },
+                {
+                    "severity": "high",
+                    "when": "2019-04-01T15:11:41.000Z",
+                    "threat": "Eicar-AV-Test",
+                    "event_service_event_id": "30acdaed-5387-4a39-b7cd-d7b9ac2a8c0f",
+                    "id": "30acdaed-5387-4a39-b7cd-d7b9ac2a8c0f",
+                    "info": None,
+                    "created_at": "2019-04-01T15:11:41.253Z",
+                    "customer_id": "816f36ee-dd2e-4ccd-bb12-cea766c28ade",
+                    "threat_cleanable": False,
+                    "data": {
+                        "created_at": 1554131501192,
+                        "endpoint_id": "c80c2a87-42f2-49b2-bab7-5031b69cd83e",
+                        "endpoint_java_id": "c80c2a87-42f2-49b2-bab7-5031b69cd83e",
+                        "endpoint_platform": "windows",
+                        "endpoint_type": "server",
+                        "event_service_id": "30acdaed-5387-4a39-b7cd-d7b9ac2a8c0f",
+                        "inserted_at": 1554131501192,
+                        "source_info": {
+                            "ip": "10.1.39.32"
+                        },
+                        "threat_id": "5ca22a2c352ea40df121ea9c",
+                        "user_match_id": "5ca22a0de5a7400deb1ab0bb"
+                    },
+                    "type": "Event::Endpoint::Threat::CleanupFailed",
+                    "location": "mock_Mercury_1",
+                    "description": "Manual cleanup required: 'Eicar-AV-Test' at 'C:\\Program Files (x86)\\Trojan Horse\\bin\\eicar.com'",
+                    "source": "n/a"
+                }
+            ]
+        }
+
+        alerts = []
+        # Run
+        try:
+            with mock.patch('siem.request_url') as mock_request_url:
+                mock_request_url.return_value = json.dumps(mock_alert_response)
+                for a in siem.call_endpoint(mock.Mock(), siem.ALERTS_V1, False,
+                                            False, 'siem_lastrun_alerts.obj', mock_config_token):
+                    alerts.append(a)
+
+        except Exception as ex:
+            print ex
+
+        # Verify
+        self.assertEqual(len(alerts), 3)
+        self.assertEqual(alerts[0]["datastream"], siem.ALERT_TYPE)
+
+        # TearDown
+        if os.path.exists('siem_lastrun_alerts.obj'):
+            os.remove('siem_lastrun_alerts.obj')
+
+
+# Run this file directly to run just the unit tests herein.
+if __name__ == "__main__":
```

Comment:
[NIT] If this file can't be run directly, it shouldn't have a main block.

Reply:
Done